featuretools/tests/primitive_tests/natural_language_primitives_tests/test_num_words.py

```diff
@@ -0,0 +1,76 @@
+import numpy as np
+import pandas as pd
+
+from featuretools.primitives import NumWords
+from featuretools.tests.primitive_tests.utils import (
+    PrimitiveT,
+    find_applicable_primitives,
+    valid_dfs,
+)
+
+
+class TestNumberOfUniqueWords(PrimitiveT):
+    primitive = NumWords
+
+    def test_general(self):
+        x = pd.Series(
+            [
+                "test test test test",
+                "test TEST test TEST,test test test",
```

Comment:
is `TEST,test` considered one word here? Is that really what we want?

Reply:
Hmm, I was under the impression that we decided not to split on punctuation in the discussion for an earlier primitive.

Reply:
Perhaps. `test,test` seems like two words to me though since I don't know of any words that contain a comma. It's probably hard to cleanly handle 100% of the cases though. As long as our docstring clearly indicates that we don't consider punctuation as a separator between words, only spaces, I think we can leave this as it is.